dotnet/Makefile

```diff
@@ -1,18 +1,24 @@
-include default.mk
+schemas = $(shell find ../jsonschema -name "*.json")
 
-.generated: .fixprotoc
+.DEFAULT_GOAL = help
 
-.fixprotoc:
-ifdef ALPINE
-	# The protoc bundled in the grpc.tools nuget package doesn't work on Alpine,
-	# so we overwrite it with the one bundled inside the Docker image.
-	# In order to prevent nuget from overwriting it again, we first
-	# download the nuget packages, then we do the copy.
-	dotnet restore
-	mkdir -p ~/.nuget/packages/grpc.tools/1.20.1/tools/linux_x64
-	cp /usr/bin/protoc ~/.nuget/packages/grpc.tools/1.20.1/tools/linux_x64/protoc
-endif
-	touch $@
+help: ## Show this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make <target>\n\nWhere <target> is one of:\n"} /^[$$()% a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
-clean:
-	rm -f Cucumber.Messages/Messages.cs .fixprotoc
+generate: require .generate-messages ## Generate c# code based on the schemas found in ../jsonschema and using the scripts in ../jsonschema/scripts for the generation
```

Comment:
Now that dotnet is supported, you can also add it to the supported languages in `Makefile` in the root of the project.